### PR TITLE
reference: add a description of max-index-length

### PR DESCRIPTION
**Pull request by zimulala**
### What is changed, added or deleted? (Required)
Add a description of `max-index-length`.
Related to pingcap/tidb#15012.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [x] v3.1 (TiDB 3.1 versions)
- [x] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

**If you select two or more versions from above**, to trigger the bot to cherry-pick this PR to your desired release version branch(es), you **must** add corresponding labels such as **needs-cherry-pick-4.0**, **needs-cherry-pick-3.1**, **needs-cherry-pick-3.0**, and **needs-cherry-pick-2.1**.

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### CODE REVIEW

**Comment by TomShawn** [comment]
LGTM
@zimulala Please involve a technical review.

**Comment by zimulala** [comment]
PTAL @bb7133 @AilinKid

File: reference/configuration/tidb-server/configuration-file.md [hunk 1]
```diff
@@ -111,6 +111,18 @@ TiDB 配置文件比命令行参数支持更多的选项。你可以在 [config/
...
 
+### `max-server-connections`
+
++ TiDB 中同时允许的最大客户端连接数，用于资源控制。
++ 默认值：0
++ 默认情况下，TiDB 不限制客户端连接数。当本配置项的值大于 `0` 且客户端连接数到达此值时，TiDB 服务端将会拒绝新的客户端连接。
+
+### `max-index-length`
+
++ 用于设置新建索引的长度限制。
```
Comment (AilinKid) [inline reference/configuration/tidb-server/configuration-file.md:122]: should put the difference with MySQL's link here?

Comment (bb7133) [reply reference/configuration/tidb-server/configuration-file.md:122]: +1, we can add a little more explainations here.

Comment (zimulala) [reply reference/configuration/tidb-server/configuration-file.md:122]: MySQL does not have this item.

Comment (zimulala) [reply reference/configuration/tidb-server/configuration-file.md:122]: @bb7133 @AilinKid PTAL

Comment (AilinKid) [reply reference/configuration/tidb-server/configuration-file.md:122]: nice work

File: reference/configuration/tidb-server/configuration-file.md [hunk 2]
```diff
@@ -111,6 +111,18 @@ TiDB 配置文件比命令行参数支持更多的选项。你可以在 [config/
...
++ 默认值：3072
++ 单位：byte，目前的合法值范围 [3072, 3072*4]。MySQL 和 TiDB v3.0.11 之前版本没有此配置项，不过都对新建索引的长度做了限制。MySQL 对此的长度限制为 3072，TiDB 在 v3.0.7 以及之前版本此值为 3072*4。为了与 MySQL 和 TiDB 之前版本的兼容，添加了此配置项。
```
Comment (TomShawn) [inline reference/configuration/tidb-server/configuration-file.md:124]: v3.0.11 and v3.0.7, the versions seem inconsistent. If this is correct, I may wonder:
+ what's the length limit in versions between v3.0.7 and v3.0.11.
+ when this limit changed from 3072*4 to 3072.

Comment (zimulala) [reply reference/configuration/tidb-server/configuration-file.md:124]: > what's the default length limit between v3.0.7 and v3.0.11.

3072.

> when this limit (default value) changed from 3072*4 to 3072.

3.0.11

Comment (TomShawn) [reply reference/configuration/tidb-server/configuration-file.md:124]:
```suggestion
+ 单位：byte
+ 目前的合法值范围 `[3072, 3072*4]`。MySQL 和 TiDB v3.0.11 之前版本（不包含 v3.0.11）没有此配置项，不过都对新建索引的长度做了限制。MySQL 对此的长度限制为 `3072`，TiDB 在 v3.0.7 以及之前版本该值为 `3072*4`，在 v3.0.7 之后的该值为 `3072`。为了与 MySQL 和 TiDB 之前版本的兼容，添加了此配置项。
```

Comment (zimulala) [reply reference/configuration/tidb-server/configuration-file.md]: @TomShawn Done.

**Pull request merged**